hibernate-core/src/main/java/org/hibernate/type/descriptor/java/LocalDateJavaDescriptor.java

```diff
@@ -112,7 +114,15 @@ public LocalDate fromString(String string) {
 
 		if ( Date.class.isInstance( value ) ) {
 			if ( java.sql.Date.class.isInstance( value ) ) {
-				return ((java.sql.Date) value).toLocalDate();
+				final java.sql.Date date = (java.sql.Date) value;
+				// If the JDBC timezone is given, the date might have been adjusted accordingly if we have a different default timezone
+				if ( options.getJdbcTimeZone() != null ) {
+					return Instant.ofEpochMilli( date.getTime() )
+							.atZone( options.getJdbcTimeZone().toZoneId() )
+							.toLocalDate();
+				} else {
+					return date.toLocalDate();
+				}
 			}
 			else {
 				return Instant.ofEpochMilli( ((Date) value).getTime() ).atZone( ZoneId.systemDefault() ).toLocalDate();
```

Comment:
Here we consider the `ZoneId.systemDefault()` instead of `options.getJdbcTimeZone()`, so there is an inconsistency which we need to resolve. I guess that the `options.getJdbcTimeZone()` should only be used by the JDBC descriptors while the `ZoneId.systemDefault()` should be used for everything else since it's also used by the `java.util.Date` when calling `toString()` or by other Local Date/Time types.